src/Executor/Promise/Adapter/ReactPromiseAdapter.php

```diff
@@ -76,15 +75,17 @@ public function createRejected($reason)
     /**
      * @inheritdoc
      */
-    public function all(array $promisesOrValues)
+    public function all(array $promisesOrValues): Promise
     {
         // TODO: rework with generators when PHP minimum required version is changed to 5.5+
```

Comment:
maybe OT but what about the todo?

Reply:
Not quite sure how the code should be changed and if there is any benefit to be had.